presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcTableHandle.java

```diff
@@ -36,19 +38,29 @@ public final class JdbcTableHandle
     private final String schemaName;
     private final String tableName;
 
+//    joinTables holds all the table handles for the same connector which are participating for the join push down
+//    this is used later in query builder to rebuild the join query
+
+    private Optional<List<ConnectorTableHandle>> joinTables;
```

Comment:
Use a `List<ConnectorTableHandle>> joinTables`. An empty list is equivalent to an `Optional.empty()` ?